### ISSUE TITLE
Create Multicall Transaction Proposal Form

### ISSUE THREAD

**Issue by skuhlmann** [HausDAO/daohaus-monorepo-deprecated#161]
# Overview

A proposal form allowing the input of any number of chained, arbitrary transaction to be executed upon a passed proposal.

This is known as the TX Builder in the v2 app.

## Use cases

A connected user can create a multi-action proposal
- open proposal form
- add basic proposal metadata/details
- add and subtract transactions 
- add details to each transaction (fields below)
- submit the proposal

All other use cases for viewing and acting on a proposal are the same as other proposals.

## Form details/fields

- Name
- Description
- Content url/link
- Proposal expiration time

Each TX:
- contract address
   - when this is added the app will query for the abi and display available methods on the contract
- function/action drop down with function names
- when a function is selected the argument fields will be populated
    - these can be a wide range of types (number, string, array, ect....)
- value - this is the amount in network token that is send to the contract from the baal

## Form Considerations

Should we allow for raw hex data to be added?
- might be helpful if abi isn't found
- but then displaying what the tx will do could be difficult and present a security risk

Might be worth tracking down as many arg types as we can so we can ensure we have designs for all types.

current design
![Screenshot from 2022-05-16 14-14-00](https://user-images.githubusercontent.com/6923345/168674841-314a967b-2edc-45de-84ab-df0e20c4a6e1.png)

**NEW CONSIDERATION**
- We now need to provide an estimated gas cost for executing the proposal action with each proposal. Our plan is to use the gnosis api behind the scenes to get that estimation.
  - do we want to display that to the user?
  - let them adjust that?
  - let them select 0, which, I think will override the estimation and let the action execute with no gas limit

## Prototype Items to Capture

**Comment by skuhlmann** [comment]
https://www.figma.com/file/9lCwprCNaoAmUDmiMQpFEA/DAOhaus-v3?node-id=7711%3A192742

**Comment by earth2travis** [comment]
From @7helles

When Jord and I talked about this a couple weeks ago, my impression was it would probably fit in a form builder app that he has been mulling over. However, since that may get pushed out an indeterminate amount of time, there are designs that could be used as a standalone proposal form version of multicall transactions: https://www.figma.com/file/9lCwprCNaoAmUDmiMQpFEA/DAOhaus-v3?node-id=7711%3A192742

**Comment by 7helles** [comment]
After the discussion yesterday about separating out the proposal-related part of transaction builder from the workbench part, I limited the scope to the workflow for building transactions from scratch and hid all icons for bookmarking/downloading.

Figma link: https://www.figma.com/file/9lCwprCNaoAmUDmiMQpFEA/DAOhaus-v3?node-id=8195%3A224354 

Notes:
- I'd originally intended the trash icon in the Transaction Overview section to clear the whole form, but I don't think it would be intuitive, so it should just clear the overview section.
- Add Expiration Date and its conditionally displayed fields can work just like in the other Proposal forms.
- I think the sections could auto-collapse once the user goes to a new section (with the actions, this would be clicking the "Add Another Action" button) as long as no required fields are empty.
- I'm not sure if we're using hr's anywhere already; if we are we can use the color number value already in use. If not, I used indigo/dark/7.
- Having collapsible sections seems nice especially if there are many action sections, but I consider that lower priority, especially in the success receipt view.
